rfcs/NNNN-cdevents/README.md

```diff
@@ -0,0 +1,126 @@
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this RFC. Describe why the change is important and the benefits to users.
+-->
+
+### Goals
+
+Integrate CDEvents into Flux with a CDEvents Receiver.
+<!--
+List the specific goals of this RFC. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+A CDEvent provider will be handled as a separate RFC in the future.
+
+<!--
+What is out of scope for this RFC? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Add CDEvents to the list of available receivers in Flux Notification controller. Similar to other receivers such as the github, the user will be able to specify the event types that causes Flux to act on inside the spec. The receiver will also verify using the [CDEvents Go SDK](https://github.com/cdevents/sdk-go) that the payload sent to the webhook URL is a valid CDEvent.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation.
+
+If the RFC goal is to document best practices,
+then this section can be replaced with the actual documentation.
+-->
+
+### User Stories
+
+<!--
+Optional if existing discussions and/or issues are linked in the motivation section.
+-->
+
+### Alternatives
+
+Certain use cases for CDEvents could be done alternatively using available receivers such as the generic webhook.
+
+<!--
+List plausible alternatives to the proposal and explain why the proposal is superior.
+
+This is a good place to incorporate suggestions made during discussion of the RFC.
+-->
+
+## Design Details
+
+Adding a receiver for CDEvents that works much like the other event-based receivers already implemented. The user will be able to write a yaml file for the receiver and deploy it to their cluster. The receiver takes the payload sent to the webhook URL, checks the headers for the event type, and filters out events based on the user-defined list of events in spec.Events. If left empty, it will act on all valid CDEvents. It then validates the payload body using the [CDEvents Go SDK](https://github.com/cdevents/sdk-go). Valid events will then trigger reconciliation of the resources.
+
+Example Receiver YAML:
+
+```yaml
+apiVersion: notification.toolkit.fluxcd.io/v1
+kind: Receiver
+metadata:
+  name: cdevents-receiver
+  namespace: flux-system
+spec:
+  type: cdevents
+  events:
+    - "dev.cdevents.change.merged"
+  secretRef:
+    name: receiver-token
+  resources:
+    - kind: Kustomization
+      apiVersion: kustomize.toolkit.fluxcd.io/v1
+      name: podinfo
+      namespace: flux-system
```

Comment:
We should use here a GitRepository, triggering the Kustomization would do nothing. Receivers are for telling Flux to pull changes from upstream sources, then Flux knows how to notify the applies (Kustomizations and HelmReleases).

Reply:
For clarity, this is how Flux eventing works
<img width="849" alt="Screenshot 2024-01-25 at 13 45 16" src="https://github.com/fluxcd/flux2/assets/3797675/5be3750c-1025-474f-bfcd-524fa03197f0">